### PR TITLE
Port 3085 changes

### PR DESCRIPTION
**Pull request by brendanww**

### CODE REVIEW

**Review by Text-only — COMMENTED** [review]
[Ether](@EtherDROPS4_bot)

**Comment by Text-only** [comment]
[ASAP](https://github.com/walletlink/walletlink/pull/169#pullrequestreview-748391828)

**Comment by Hlllllllii** [comment]
Alex Siri . W , m . Try it.


Sent from Yahoo Mail for iPhone


On Tuesday, September 7, 2021, 3:50 PM, DANIEL RAMER ***@***.***> wrote:




@Text-only commented on this pull request.

Ether

File: js/src/relay/WalletLinkRelay.ts [hunk 1]
```diff
@@ -517,8 +573,8 @@ export class WalletLinkRelay implements WalletLinkRelayAbstract {
     message: RelayMessage,
     callWebhook: boolean
   ): Observable<string> {
-    const secret = this.session.secret
-    return new Observable<string>((subscriber) => {
+    let secret = this.session.secret
+    return new Observable<string>(function (subscriber) {
```
Comment (vishnumad) [inline js/src/relay/WalletLinkRelay.ts:577]: Can you change this back? Doesn't look like secret is being reassigned.

**Review by vishnumad — APPROVED** [review]

File: js/package.json [hunk 2]
```diff
@@ -17,8 +17,8 @@
     "walletlink",
     "web3"
   ],
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "build/npm/dist/index.js",
```
Comment (hieronymus777) [inline js/package.json:20]: I recall that this change wasn't necessary in the past - wondering why it's here

Comment (hieronymus777) [reply js/package.json:20]: It's also not in the original pr

Comment (brendanww) [reply js/package.json:20]: We don't build to the `dist` folder anymore. Needed to make this change in order to successfully test local test dapp against these changes.

**Review by hieronymus777 — APPROVED** [review]

File: js/src/relay/WalletLinkRelay.ts (hunk 1 above)
Comment (godspower77) [reply js/src/relay/WalletLinkRelay.ts]: Blockhan wellet address: 90aa4a2b-3fc3-449c-9492-a4142b3fbda6

**Comment by Peiman25800** [comment]
sadeghzare916

در تاریخ چهارشنبه ۸ سپتامبر ۲۰۲۱،‏ ۰۱:۲۰ DANIEL RAMER <
***@***.***> نوشت:

**Comment by godspower77** [comment]
Blockhan wellet ID: 90aa4a2b-3fc3-449c-9492-a4142b3fbda6
Backup: issue balcony jewel expand cement great draft filter general actual borrow pride

**Pull request merged**